src/systems/environmental_sensor_system/EnvironmentalSensorSystem.cc

```diff
@@ -76,19 +78,115 @@ class EnvironmentalSensor : public gz::sensors::Sensor
     // Load common sensor params
     gz::sensors::Sensor::Load(_sdf);
 
-    this->pub = this->node.Advertise<gz::msgs::Double>(this->Topic());
+    // Handle the field type
+    if (_sdf.Element() != nullptr &&
+      _sdf.Element()->HasElement("field_type"))
+    {
+      auto dataType = _sdf.Element()->Get<std::string>("field_type");
+      if (dataType == "scalar")
+      {
+        this->numberOfFields = 1;
+      }
+      else if (dataType == "vector2")
+      {
+        this->numberOfFields = 2;
+      }
+      else if (dataType == "vector3")
+      {
+        this->numberOfFields = 3;
+      }
+      else
+      {
+        gzerr << "Invalid data type of " << dataType << " given."
+          << "valid options are scalar, vector2 or vector3. "
+          << "Defaulting to scalar" << std::endl;
+      }
+    }
+
+    // Hansle the transform type
+    if (_sdf.Element() != nullptr &&
+      _sdf.Element()->HasElement("transform_type"))
+    {
+      auto res =
+        getTransformType(_sdf.Element()->Get<std::string>("transform_type"));
+
+      if (!res.has_value())
+      {
+        gzerr << "Invalid transform type of "
+          << _sdf.Element()->Get<std::string>("transform_type")
+          << " given. Valid types are ADD_VELOCITY_LOCAL, ADD_VELOCITY_GLOBAL,"
+          << " LOCAL, and GLOBAL. Defaulting to GLOBAL." << std::endl;
+      }
+      else
+      {
+        this->transformType = res.value();
+      }
+    }
+
+    switch(this->numberOfFields) {
+      case 1:
+        this->pub = this->node.Advertise<gz::msgs::Double>(this->Topic());
+        break;
+      case 2:
+        this->pub = this->node.Advertise<gz::msgs::Vector2d>(this->Topic());
+        break;
+      case 3:
+        this->pub = this->node.Advertise<gz::msgs::Vector3d>(this->Topic());
+        break;
+      default:
+        gzerr << "Unable to create publisher too many fields" << "\n";
+    }
 
     // If "environment_variable" is defined then remap
     // sensor from existing data.
     if (_sdf.Element() != nullptr &&
+      this->numberOfFields == 1 &&
       _sdf.Element()->HasElement("environment_variable"))
     {
-      this->fieldName =
+      this->fieldName[0] =
         _sdf.Element()->Get<std::string>("environment_variable");
     }
+    else if (_sdf.Element() != nullptr &&
+      this->numberOfFields == 2)
+    {
+      if (
+        _sdf.Element()->HasElement("x_variable") &&
```

Comment:
@arjo129 meta: I find tag naming a bit inconsistent (`environment_variable` vs. `x_variable`, `y_variable`). Considering that variable names are likely to follow a pattern (e.g. `<name>_<component>`), why not enforce a convention? Then we wouldn't need to specify each component, only the `environment_variable` e.g. `wind_speed`.

This will also keep the interface future-proof for when file formats that natively support vectorial data are added (like netCDF).

Reply:
Hmm how would that work with our current data? The CSV column headers would be fixed?

Reply:
Oh ok I get it you would define the `wind_speed_x` and the `wind_speed_y` in the CSV. We would need to change the header of our CSV files to match and then just specify `<environment_variable>wind_speed</environment_variable>` in the SDFormat file.

Reply:
Hmmm... I just checked the DVL it seems we have the x_column, y_column and z_column. I'd like this to be consistent with it.

Reply:
IMO it's not an exact analogue because the DVL will always work with 3D data whereas this has to deal with various dimensions (and even then perhaps the DVL should change to use multi-dimensional variables if supported).

That said, in the interest of time, I'm OK with not switching to a convention yet, but at least tag naming should be consistent across dimensions (i.e. it should follow some pattern and not be completely different if it's 1D or 2D).

Reply:
In the interest of time I'm keeping `x_variable`, `y_variable` and `z_variable`. When NetCDF is supported we could just use `environment_variable`